include/zephyr/llext/loader.h

```diff
@@ -73,7 +73,7 @@ struct llext_loader {
 	elf_shdr_t sects[LLEXT_MEM_COUNT];
 	elf_shdr_t *sect_hdrs;
 	bool sect_hdrs_on_heap;
-	enum llext_mem *sect_map;
+	struct llext_elf_sect_map *sect_map;
```

Comment:
There was one in the first draft, I removed it on purpose - C standard explicitly allows you to declare opaque pointers to structures this way. But it makes sense to leave a comment referring to the file, thanks for spotting this!

Reply:
@pillo79 wow, which C version explicitly allows that? I only know about compilers warning about such missing declarations

Reply:
This is apparently in the language since the first standard (C89/C90) if [cppreference](https://en.cppreference.com/w/c/language/struct) is to be trusted :exploding_head: 

I was completely unaware of this as well, but it appears to indeed be supported fine; GCC 3.4.6 from March 2006 accepts it without problem, even in `-std=c89` mode, [according to Compiler Explorer](https://godbolt.org/z/Eh8bfP7Kx).

Reply:
Yes, it's plain C - it's the reason why you may get a weird "this type is only usable inside the function" message from your compiler when you make a typo in a `struct` argument during a function call definition. 

That type is really an implementation detail, and I wanted to hide it as much as possible - since it's declared in a `@cond` block, it will not even be picked up by documentation. I will add the forward declaration and explicitly hide it, but it looked ugly and redundant to me.

Reply:
> Yes, it's plain C - it's the reason why you may get a weird "this type is only usable inside the function" message from your compiler when you make a typo in a `struct` argument during a function call definition.

hm, interesting. So you can use a pointer to an incomplete type in a structure, but not a function parameter. And forward declarations are only needed for the latter? Even though I haven't found a definitive answer apart from
> Note that a new struct name may also be introduced just by using a struct tag within another declaration

in https://en.cppreference.com/w/c/language/struct but yes, looks like that's the case. I stand corrected, learned something new, thanks!